File: app/src/main/java/net/borkiss/randomuser/data/UserDataSource.java
package net.borkiss.randomuser.data;

import androidx.annotation.NonNull;

import net.borkiss.randomuser.data.model.User;

import java.util.List;

public interface UserDataSource {

    interface LoadUsersCallback {
        void onUsersLoaded(List<User> users);
        void onDataNotAvailable();
    }

    interface GetUserCallback {
        void onUserLoaded(User user);
        void onDataNotAvailable();
    }

    void getAllUsers(@NonNull LoadUsersCallback callback);

}

File: app/src/main/java/net/borkiss/randomuser/data/UserDataSource.kt
package net.borkiss.randomuser.data

import net.borkiss.randomuser.data.model.User

interface UserDataSource {

    interface LoadUsersCallback {
        fun onUsersLoaded(users: List<User>)
        fun onDataNotAvailable()
    }

    fun getAllUsers(callback: LoadUsersCallback)
}
